### PR TITLE
feat(relay): Add spooling configuration options

### PR DESCRIPTION
**Pull request by olksdr**
Adding new spooling configuration options:
* path to the file
* max memory size
* max disk size
* max / min connections

### CODE REVIEW

File: src/docs/product/relay/options.mdx [hunk 1]
```diff
@@ -199,6 +199,42 @@ See also `cache.envelope_expiry`.
...
 
+## Spooling
+
+These settings control on-disk spooling. Currently only spooling for incoming envelopes is supported.
+
+`spool.envelopes.path`
+
+: _String, default: null_
+
+  The path to the spool DB file, which is used to persist the envelopes to disk.
```
Comment (iker-barriocanal) [inline src/docs/product/relay/options.mdx:210]: nit: I think DB is too much details for this.
```suggestion
  The path to the spool file, which is used to persist the envelopes to disk.
```

File: src/docs/product/relay/options.mdx [hunk 2]
```diff
@@ -199,6 +199,42 @@ See also `cache.envelope_expiry`.
...
+  If provided enables on disk spooling.
+
+`spool.envelopes.max_memory_size`
+
+: _String, default: `500MB`_
```
Comment (iker-barriocanal) [inline src/docs/product/relay/options.mdx:215]: Units are separated from the value (see examples above), so either
```suggestion
: _String, default: `500` (MB)_
```
or
```suggestion
: _String, default: `500` (500MB)_
```
Same with the unit below.

**Review by iker-barriocanal — APPROVED** [review]
Please fix the unit before merging.

File: src/docs/product/relay/options.mdx (hunk 2 above)
Comment (olksdr) [reply src/docs/product/relay/options.mdx:215]: Provided values are correct, see for example how the `limits` below configured. 
This is the format (human readable) which is used to configure this options.

File: src/docs/product/relay/options.mdx (hunk 1 above)
Comment (lizokm) [inline src/docs/product/relay/options.mdx:210]:
```suggestion
These settings control on-disk spooling. Currently, only spooling for incoming envelopes is supported.

`spool.envelopes.path`

: _String, default: null_

  The path to the spool DB file, which is used to persist the envelopes to disk.
```
Not familiar with this concept, but wan to double check that "persist" is the correct word to be used here?

File: src/docs/product/relay/options.mdx [hunk 3]
```diff
@@ -199,6 +199,42 @@ See also `cache.envelope_expiry`.
...
+## Spooling
+
+These settings control on-disk spooling. Currently only spooling for incoming envelopes is supported.
+
+`spool.envelopes.path`
+
+: _String, default: null_
+
+  The path to the spool DB file, which is used to persist the envelopes to disk.
+  If provided enables on disk spooling.
```
Comment (lizokm) [inline src/docs/product/relay/options.mdx:211]:
```suggestion
  If provided, enables on-disk spooling.
```

File: src/docs/product/relay/options.mdx [hunk 4]
```diff
@@ -199,6 +199,42 @@ See also `cache.envelope_expiry`.
...
+: _String, default: `500MB`_
+
+  The maximum in-memory buffer for incoming envelopes.
+  The buffer will be spooled to disk when the configured size is reached.
+
+`spool.envelopes.max_disk_size`
+
+: _String, default: `500MB`_
+
+  The maximum size on disk spool.
```
Comment (lizokm) [inline src/docs/product/relay/options.mdx:224]:
```suggestion
  The maximum size on-disk spool.
```

**Review by lizokm — APPROVED** [review]
A few tiny tweaks :)

File: src/docs/product/relay/options.mdx (hunk 1 above)
Comment (olksdr) [reply src/docs/product/relay/options.mdx:210]: Thanks! Changed it to `save` , which precisely describes what's going on.

**Pull request merged**